project_euler/problem_073/sol1.py

```diff
@@ -0,0 +1,40 @@
+"""
+Project Euler Problem 73: https://projecteuler.net/problem=73
+
+Consider the fraction, n/d, where n and d are positive integers.
+If n<d and HCF(n,d)=1, it is called a reduced proper fraction.
+
+If we list the set of reduced proper fractions for d ≤ 8 in ascending order of size,
+we get:
+
+1/8, 1/7, 1/6, 1/5, 1/4, 2/7, 1/3, 3/8, 2/5, 3/7, 1/2, 4/7, 3/5, 5/8, 2/3,
+5/7, 3/4, 4/5, 5/6, 6/7, 7/8
+
+It can be seen that there are 3 fractions between 1/3 and 1/2.
+
+How many fractions lie between 1/3 and 1/2 in the sorted set
+of reduced proper fractions for d ≤ 12,000?
+"""
+
+from math import gcd
+
+
+def solution(max_d: int = 12_000) -> int:
+    """
+    Returns number of fractions lie between 1/3 and 1/2 in the sorted set
+    of reduced proper fractions for d ≤ max_d
+
+    >>> solution(8)
```

Comment:
We should add more tests. Examples: when output is 0, 1 and maybe like > 20 or something.

Reply:
Added tests when output is 0, 1. Good test (when output is quite large) is checked by `validate-solutions` script